copy/copy.go

```diff
@@ -577,6 +577,29 @@ func (c *copier) copyOneImage(ctx context.Context, policyContext *signature.Poli
 		}
 	}
 
+	destExistsCheck := true // TBD: enable configuration
+	if destExistsCheck {
+		destImageSource, err := c.dest.Reference().NewImageSource(ctx, options.DestinationCtx)
+
+		if err == nil {
+
+			retManifest, retManifestType, err = destImageSource.GetManifest(ctx, nil)
+			if err != nil {
+				return nil, "", "", errors.Wrapf(err, "Error reading manifest from dest image %s", transports.ImageName(destImageSource.Reference()))
```

Comment:
```suggestion
				return nil, "", "", errors.Wrapf(err, "Error reading manifest from destination image %s", transports.ImageName(destImageSource.Reference()))
```

Reply:
BTW most of the code in this file does not say source/destination at all, the image name + operation should be sufficient.

Reply:
… but anyway we need to carefully deal with correctness first, and see what text is necessary for that, before fine-tuning it.

(e.g. — note to self — we still need to do the copy if the source contains signatures. There may be more.)